.github/workflows/coverage_run_and_upload.yml

```
@@ -39,15 +42,15 @@ jobs:
               do echo "Generating test coverage for $lib" && cd "$lib" && go mod tidy && go test -v -coverpkg=./... -cover -covermode atomic -coverprofile $TRACER_PATH/coverage/coverage_$(date +%s%N)_$RANDOM.out ./... -json > $TRACER_PATH/coverage/coverage_$(date +%s%N)_$RANDOM.json && cd -;
             done
 
-            echo "Generating test coverage for ./instrumentation/instagocb"
-            cd ./instrumentation/instagocb
-            go mod tidy
-            go test -v -tags=integration -coverpkg=./... -cover -covermode atomic -coverprofile $TRACER_PATH/coverage/coverage_$(date +%s%N)_$RANDOM.out ./... -json > $TRACER_PATH/coverage/coverage_$(date +%s%N)_$RANDOM.json && cd -
+            INTEGRATIONS_TESTS=("instagocb" "instapgx" "instacosmos")
 
-            echo "Generating test coverage for ./instrumentation/instapgx"
-            cd ./instrumentation/instapgx
-            go mod tidy
-            go test -v -tags=integration -coverpkg=./... -cover -covermode atomic -coverprofile $TRACER_PATH/coverage/coverage_$(date +%s%N)_$RANDOM.out ./... -json > $TRACER_PATH/coverage/coverage_$(date +%s%N)_$RANDOM.json && cd -
+            for str in ${INTEGRATIONS_TESTS[@]}; do
+              dir=./instrumentation/"$str"
```

Feel free to correct if if I'm wrong but I dont think these quotes are needed here. 

```suggestion
              dir=./instrumentation/$str
```

---

It's correct. Good catch